src/dom/Draggable.js

```diff
@@ -44,7 +43,8 @@ export var Draggable = Evented.extend({
 		// @option clickTolerance: Number = 3
 		// The max number of pixels a user can shift the mouse pointer during a click
 		// for it to be considered a valid click (as opposed to a mouse drag).
-		clickTolerance: 3
+		clickTolerance: 3,
+		mapContainer: undefined

```

Comment:
does it mean that mapContainer is required for every single usage of `Draggable`?

Reply:
@cherniavskii not required, but optional. if you haven't specified it nothing shall be broken. But i must give container instance to getMousePosition, it must have access to it. haven't other idea how to have that access in better way.

Reply:
As @yarsky-tgz figured out, in order to have `Draggable` correctly compute the drag offset, taking into account the map container CSS scale, it must have a reference somehow to that map container. This is easy for `Map.Drag`, but not for `Marker.Drag`.

With the current PR state, this new `mapContainer` option is indeed optional.
`Draggable` behaves as previously when this option is not set (i.e. it uses the pointer position directly on screen).

Thinking about it further, this shows that `Draggable` parameters are simply not enough:
1. Moving element
2. Drag target
3. (prevent outline)
4. (options)

…but what about the drag area, which is potentially affected by a CSS scale?
Again, in the case of `Map.Drag`, we simply have Drag target === drag area, but that is not the case for `Marker.Drag` (the `icon` can have its own CSS scale).

Therefore adding a new parameter or option is necessary in order to be able to take into account the container CSS scale. Maybe a more generic `dragArea` or `dragContainer` name could be used?

Reply:
…or maybe "`dragScaledContainer`", to make it more obvious that this option is used to retrieve the CSS scale and correct the computed movement accordingly, rather than the area on which the drag is effective (drag does not stop if user points outside the dragTarget, or whatever container).

No reference to "map" is needed, since `Draggable` can be applied to any arbitrary Element, not necessarily something related to a Leaflet map:
https://plnkr.co/edit/m2LUcgmcbOSvliuANumg?p=preview

Reply:
All right, there should be an interesting workaround that would enable getting rid of the `mapContainer` option.

What we need is to retrieve the CSS scale which applies on the dragged element (i.e. `this._element` in `Draggable` instance), except for a potential scale on the very element itself. This scale is determined by the combined scales of the hierarchy of DOM parent nodes of the dragged element. Therefore reading the scale of the element's `parentNode` should be enough to compute the scale correction.

This works fine for the map, but for a draggable Marker, the difficulty is that the `leaflet-marker-pane` is of size `0`, which makes the scale computation algorithm (as implemented here) fail. So we can walk up the DOM parent nodes hierarchy until we find an element of size !== 0, crossing fingers that intermediate elements maintain the CSS scale. That is the case for the `leaflet-map-pane` and the `leaflet-marker-pane`.
```javascript
function getNonNullSizeParentNode(element) {
  do {
    element = element.parentNode;
  } while ((!element.offsetWidth || !element.offsetHeight) && element !== document.body)
  
  return element;
}
```
Example: https://plnkr.co/edit/HYyDssl2F2rpvwmDYuW6?p=preview